mistlib/core.py

```diff
@@ -356,7 +356,145 @@ def append_file(input_filename, output_filename):
+        thermal_expansion = self.get_property("thermal_expansion", code_name, reference_temperature)
+        
+        with open(file, "w") as f:
+            f.write(comment_block)
+            f.write("solid\n{\n")
+            
+            p = self.properties["thermal_conductivity_solid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tkappa\t ({thermal_cond_solid[0]} {thermal_cond_solid[1]} 0.0);\n")
+            else:
+                f.write(f"\tkappa\t ({self.properties['thermal_conductivity_solid'].value} 0.0 0.0);\n")
+
+            p = self.properties["specific_heat_solid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tCp\t\t ({specific_heat_solid[0]} {specific_heat_solid[1]} 0.0);\n")
+            else:
+                f.write(f"\tCp\t\t ({self.properties['specific_heat_solid'].value} 0.0 0.0);\n")
+            
+            f.write("}\n\n")
+            f.write("liquid\n{\n")
 
+            p = self.properties["thermal_conductivity_liquid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tkappa\t ({thermal_cond_liquid[0]} {thermal_cond_liquid[1]} 0.0);\n")
+            else:
+                f.write(f"\tkappa\t ({self.properties['thermal_conductivity_liquid'].value});\n")
+            
+            p = self.properties["specific_heat_liquid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tCp\t\t ({specific_heat_liquid[0]} {specific_heat_liquid[1]} 0.0);\n")
+            else:
+                f.write(f"\tCp\t\t ({self.properties['specific_heat_liquid'].value} 0.0 0.0);\n")
+
+            f.write("}\n\n")
+            f.write("powder\n{\n")
+            
+            p = self.properties["thermal_conductivity_solid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tkappa\t ({thermal_cond_solid[0]} {thermal_cond_solid[1]} 0.0);\n")
+            else:
+                f.write(f"\tkappa\t ({self.properties['thermal_conductivity_solid'].value} 0.0 0.0);\n")
+
+            p = self.properties["specific_heat_solid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tCp\t\t ({specific_heat_solid[0]} {specific_heat_solid[1]} 0.0);\n")
+            else:
+                f.write(f"\tCp\t\t ({self.properties['specific_heat_solid'].value} 0.0 0.0);\n")
+            
+            f.write("}\n\n")
+            f.write(f"rho     [1 -3 0 0 0 0 0]    {density};\n"
+        f"mu      [1 -1 -1  0 0 0 0]  {dynamic_viscosity};\n"
+        f"beta    [0 0 0 -1 0 0 0]    {thermal_expansion};\n"
+        f"DAS     [0 1 0 0 0 0 0]     10e-6;\n"
+        f"Lf      [0  2 -2  0 0 0 0]  {latent_heat_fusion:.2e};\n\n"
+        f"// ************************************************************************* //")
+            
+    def write_additivefoam_thermoPath(self, file):        
```

Comment:
Should there be a top-level function `write_additivefoam_input` that calls all the other ones?

Reply:
I think it would make sense to have a top-level function to do everything you need for additivefoam.

Reply:
Working on the top level function for AdditiveFOAM and a better way to handle polynomial values for material constants.

Reply:
Added